### ISSUE TITLE
Cast the result of malloc()

### ISSUE THREAD

**Issue by bartnijssen** [UW-Hydro/VIC#247]
Apparently, casting the result of `malloc()` is not the preferred way to go. See for example:
http://stackoverflow.com/questions/605845/do-i-cast-the-result-of-malloc

The same is true for `calloc()`

We have cast religiously in VIC. With the release of VIC 5 we may as well conform to convention since the code is being reformatted anyway.

My preferred format in that case would be 

```C
mvar = malloc(n * sizeof(*mvar));
cvar = calloc(n, sizeof(*cvar));
```
 * This way there is no type info in the `malloc` or `calloc` call, which makes things more maintainable (for example if `mvar` is changed from a `float` to a `double`, no change to the `malloc` or `calloc` call is required).

 * Even though the `()` are not required after `sizeof`, it definitely improves readability (prevents confusion with the multiplication operator), so I would like to maintain this.

 * in `malloc` have the number (`n` in this case) first, followed by the size (similar to `calloc`).

 * Note the extra `*` in `sizeof()`. You want to create memory sufficient to store the value of `mvar` and `cvar`. You'll have to be a little thoughtful in case of n-dimensional arrays. It would be good to include examples of that as part of this issue.

**Comment by tbohn** [comment]
It's nice to see a solution to that problem!

**Comment by jhamman** [comment]
closed via #257
